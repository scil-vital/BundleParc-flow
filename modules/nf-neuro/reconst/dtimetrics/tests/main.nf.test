nextflow_process {

    name "Test Process RECONST_DTIMETRICS"
    script "../main.nf"
    process "RECONST_DTIMETRICS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "reconst"
    tag "reconst/dtimetrics"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "DWIss1000-dir32.zip", "segmentation.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - stub-run dtimetrics") {
        options "-stub-run"
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        dwi: it.simpleName == "DWIss1000-dir32"
                        segmentation: it.simpleName == "segmentation"
                    }
                input[0] = ch_split_test_data.dwi.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        []
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
