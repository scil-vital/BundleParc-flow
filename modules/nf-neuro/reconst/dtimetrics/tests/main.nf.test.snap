{
    "reconst - stub-run dtimetrics": {
        "content": [
            [
                "versions.yml:md5,c7a6b86e604d3b25340611cec27c758a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-06T15:57:03.763217558"
    }
}